rest_lights.cpp

```diff
@@ -1973,101 +1973,113 @@ int DeRestPluginPrivate::setWindowCoveringState(const ApiRequest &req, ApiRespon
 int DeRestPluginPrivate::setTuyaDeviceState(const ApiRequest &req, ApiResponse &rsp, TaskItem &taskRef, QVariantMap &map)
 {
     QString id = req.path[3];
+    
     bool targetOn = false;
-
-    if (map.contains("on"))
+    bool hasOn = false;
+    bool hasBri = false;
+    qint16 targetBri = 0;
+    
+    bool ok = false;
+    
+    if (map.contains("bri") && R_GetProductId(taskRef.lightNode).startsWith(QLatin1String("Tuya_DIMSWITCH")))
+    {
+        if (map["bri"].type() == QVariant::Double)
+        {
+            hasBri = true;
+            targetBri = map["bri"].toUInt(&ok);
+        }
+    }
+    if (map.contains("on") && taskRef.lightNode->item(RStateOn) )
     {
         if (map["on"].type() == QVariant::Bool)
         {
-            bool ok = false;
-            qint8 button = DP_IDENTIFIER_BUTTON_1;
-            QByteArray data;
-
+            hasOn = true;
             targetOn = map["on"].toBool();
+        }

```

Comment:
Missing error else branch to report error and return from function.

Reply:
a little better